qctrlopencontrols/dynamic_decoupling_sequences/driven_controls.py

```diff
@@ -270,6 +273,20 @@ def convert_dds_to_driven_control(
                                   extras={'deduced_pulse_start_timing': pulse_start_ends[:, 0],
                                           'deduced_pulse_end_timing': pulse_start_ends[:, 1]})
 
+    # check if the minimum_segment_duration is respected in the gaps between the pulses
+    if not np.any(np.logical_or(np.greater(pulse_start_ends[1:, 0]-pulse_start_ends[:-1, 1],
```

Comment:
Hm, I'm not sure this is right. The "or" is checking "is the gap >= minimum_segment_duration". The "any" is checking "are any of the gaps >= minimum_segment_duration". I think we want "all" instead?

Reply:
Ooops, sorry, I really messed this up. Yeah, should be `np.all` and `np.isclose` should be comparing the gap with `minimum_segment_duration` rather than with each other (I was thinking about Luigi's problem when I was writing this and I guess I got crossed wires in my brain haha). I adopted your `gap_durations` array to make things less confusing to myself, thanks